### DISCUSSION TITLE
Confused about env variables

### DISCUSSION THREAD

**Discussion by Darren120** [gladly-team/next-firebase-auth#304] [Q&A]
Hello my nextjs app doesn't read the env variable idk why and I am getting errors 

Error: Invalid next-firebase-auth options: The "firebaseClientInitConfig.apiKey" value is required.

**Comment by Darren120** [comment]
Ok, so I put NEXT_PUBLIC and now it works. will this work in production too?

**Comment by Darren120** [comment]
Is the cookies setting using the /api required for ssr?

**Comment by Darren120** [comment]
Hi, I don't like FirebaseAuth black box, this is enough black box. How do I make a custom firebaseAuth? I don't know how FirebaseAuth calls the login api

**Comment by BenPust** [comment]
This is for anyone who was also confused after getting `"firebaseClientInitConfig.apiKey" value is required` after populating the keys on the example project. Make sure to rename the env file by removing `.example` from `.env.local.example` to `.env.local`
